### PR TITLE
perf: use fast ops for stack trace remapping

### PR DESCRIPTION
**Pull request by lucacasonato**
Requires #214

### CODE REVIEW

File: core/source_map.rs [hunk 1]
```diff
@@ -39,54 +40,99 @@ where
 pub struct SourceMapCache {
   maps: HashMap<String, Option<SourceMap>>,
   source_lines: HashMap<(String, i64), Option<String>>,
+  // This is not the right place for this, but it's the easiest way to make
+  // op_apply_source_map a fast op. This stashing should happen in #[op2].
```
Comment (mmastrac) [inline core/source_map.rs:44]: 👍

**Review by mmastrac — APPROVED** [review]
LGTM, one minor comment.

**Review by littledivy — APPROVED** [review]
LGTM

File: core/ops_builtin_v8.rs [hunk 2]
```diff
@@ -813,7 +813,7 @@ pub struct Location {
 #[op2(core, fast)]
 #[smi]
 pub fn op_apply_source_map(
-  state: &mut JsRuntimeState,
+  state: &JsRuntimeState,
```
Comment (mmastrac) [inline core/ops_builtin_v8.rs:816]: Nice fix!

**Review by bartlomieju — APPROVED** [review]

**Comment by lucacasonato** [comment]
@mmastrac whoops, missed your review comments. addressed in https://github.com/denoland/deno_core/pull/222

**Pull request merged**